lucene/core/src/java/org/apache/lucene/index/CheckIndex.java

```diff
@@ -465,9 +493,22 @@ public void setChecksumsOnly(boolean v) {
   
   private boolean checksumsOnly;
 
-  /** Set infoStream where messages should go.  If null, no
-   *  messages are printed.  If verbose is true then more
-   *  details are printed. */
+  /** Set threadCount used for parallelizing index integrity checking. */
+  public void setThreadCount(int tc) {
+    if (tc <= 0) {
+      throw new IllegalArgumentException(
+          "setThreadCount requires a number larger than 0, but got: " + tc);
+    }
+    threadCount = tc;
+  }
+
+  // capped threadCount at 4 for default
+  private int threadCount = Math.min(Runtime.getRuntime().availableProcessors(), 4);
```

Comment:
I have actually updated that in the other PR for updating instruction as well (since the updated instruction also mentioned about the number of CPU cores for default) https://github.com/apache/lucene/pull/281, and plan to add that commit into this PR once it is merged. Would you like me to track this separately instead?